zeroknowledgeproof/src/main/java/com/zeroknowledgeproof/rangeProof/RangeProofVerifier.kt

```diff
@@ -0,0 +1,146 @@
+package com.zeroknowledgeproof.rangeProof
+
+import java.math.BigInteger
+import java.math.BigInteger.ONE
+import java.math.BigInteger.ZERO
+import java.security.MessageDigest
+import java.security.Security
+
+class RangeProofVerifier(private val N: BigInteger, private val low: Int, private val up: Int) {
+    private val TAG = "RangeProofVerifier"
+
+    fun requestChallenge(bound: BigInteger): Challenge {
+        // Step 5: Generate s,t in Zk1 - {0}
+        val s = generateRandomInterval(ONE, bound)
+        val t = generateRandomInterval(ONE, bound)
+        return if (s == t)
+            requestChallenge(bound)
+        else
+            Challenge(s, t)
+    }
+
+    fun interactiveVerify(setupRes: SetupPublicResult, answer: InteractivePublicResult): Boolean {
+        // Step 7: Verifier verifies ALL of these claims.
+        // First extract all structs
+        val (c, c1, c2, sameCommitment, cPrime, cDPrime, cDPrimeIsSquare, c1Prime, c2Prime, c3Prime,
+                m3IsSquare, g, h, k1) = setupRes
+
+        val (s, t) = answer.challenge
+        val (x, y, u, v) = answer
+
+        // Check that nothing is zero
+
+        return c != ZERO &&
```

Comment:
Let's put some of the conditions in the when clause in seperate functions and let's and them all together in a nicer way than this 50 line return statement :)

Reply:
Excellent point.